### PR TITLE
Eahw 589 fix the api filter functionality

### PR DESCRIPTION
**Pull request by JWeston-HO**
This PR is to address the filter functionality required for eahw-589.
The requirement is to show time entries that cover two days to show on the day they start and on the day they end.

### CODE REVIEW

File: src/utils/filters/time-entry-filter/timeEntryFilterBuilder.js [hunk 1]
```diff
@@ -2,13 +2,36 @@ import dayjs from 'dayjs';
 import {
   filterOnOrAfterDate,
   filterBeforeDate,
+  joinOrConditions,
+  joinAndConditions,
 } from '../resource-filter-builder/resourceFilterBuilder';
 
-export const filterTimeEntriesOnDate = (date) => {
-  const startOfDateFilter = filterOnOrAfterDate('actualStartTime', date);
+export const filterTimeEntriesOnDate = (resourceDateProperty, date) => {
```
Comment (daleelR) [inline src/utils/filters/time-entry-filter/timeEntryFilterBuilder.js:9]: we no longer need to export this function as `buildTimeCardFilter` is the one used by `TimeCard` now

File: src/utils/api-utils/UrlSearchParamBuilder.js [hunk 2]
```diff
@@ -10,8 +10,8 @@ export class UrlSearchParamBuilder {
     return this;
   }
 
-  setFilters(...filters) {
-    this.params.append('filter', filters.join('&&'));
+  setFilters(filter) {
```
Comment (daleelR) [inline src/utils/api-utils/UrlSearchParamBuilder.js:13]: nit: could this be named `setFilter` without the 's'? Then if we ever need to re-introduce the above removed function for multiple separate filters we can with it's appropriate name.

File: src/utils/filters/time-entry-filter/timeEntryFilterBuilder.js [hunk 3]
```diff
@@ -2,13 +2,36 @@ import dayjs from 'dayjs';
...
+  const startOfDateFilter = filterOnOrAfterDate(resourceDateProperty, date);
   const endOfDateFilter = filterBeforeDate(
-    'actualStartTime',
+    resourceDateProperty,
     dayjs(date).add(1, 'day')
   );
   return [startOfDateFilter, endOfDateFilter];
 };
+
+export const buildTimeCardFilter = (date, userId) => {
```
Comment (daleelR) [inline src/utils/filters/time-entry-filter/timeEntryFilterBuilder.js:18]: nit: should probs be named buildTime**Entries**Filter instead.

File: src/utils/filters/time-entry-filter/timeEntryFilterBuilder.js (hunk 1 above)
Comment (JWeston-HO) [reply src/utils/filters/time-entry-filter/timeEntryFilterBuilder.js]: export removed.

File: src/utils/api-utils/UrlSearchParamBuilder.js (hunk 2 above)
Comment (JWeston-HO) [reply src/utils/api-utils/UrlSearchParamBuilder.js]: Changed to setFilter

**Review by daleelR — APPROVED** [review]

**Review by LaurieBennett — APPROVED** [review]

**Pull request merged**